tests/monotouch-test/CoreFoundation/SocketTest.cs

```diff
@@ -0,0 +1,67 @@
+﻿//
+// Unit tests for CFSocket
+//
+// Authors:
+//	Marius Ungureanu <maungu@microsoft.com>
+//
+// Copyright 2019 Microsoft Inc. All rights reserved.
+//
+
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+using System.Threading;
+#if XAMCORE_2_0
+using Foundation;
+using CoreFoundation;
+using ObjCRuntime;
+#else
+using MonoTouch.CoreFoundation;
+using MonoTouch.Foundation;
+using MonoTouch.UIKit;
+#endif
+using NUnit.Framework;
+
+namespace MonoTouchFixtures.CoreFoundation {
+
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class CFSocketTest {
+		static readonly byte[] dataToSend = { 0, 1, 2, 3 };
+
+		[Test]
+		public void RetainCount ()
```

Comment:
I think it's also possible to create a test that checks if the CFSocket instance is actually collected by the GC:

1. Create several CFSockets on a background thread, create a weak gchandle to each, and then dispose them.
2. On the main thread, wait for the background thread to finish.
3. Run the GC a few times.
4. Assert that at least one of the weak gchandles doesn't have a target anymore.

This test should ensure that the refcounting logic for the context is correct.

Reply:
Did manual tracing on the unit test, it was collecting them. But sure, can add that.